griptape/common/prompt_stack/messages/prompt_stack_message.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from typing import Any
+
+from attrs import define, field
+
+from griptape.artifacts import TextArtifact
+from griptape.common import BasePromptStackContent, TextPromptStackContent
+
+from .base_prompt_stack_message import BasePromptStackMessage
+
+
+@define
+class PromptStackMessage(BasePromptStackMessage):
+    def __init__(self, content: str | list[BasePromptStackContent], **kwargs: Any):
+        if isinstance(content, str):
+            content = [TextPromptStackContent(TextArtifact(value=content))]
+        self.__attrs_init__(content, **kwargs)  # pyright: ignore[reportAttributeAccessIssue]
```

Comment:
could use an attrs `converter` here

Reply:
I don't think there's a good way to use a converter and get proper type hints. With a converter, you'll get yelled at for trying to pass in a string. And updating the field's type hints leads down a path of `Union`s everywhere.